-1

Tested at: 815d07da90cfb53606e7f050021b03c1a67a6524

CMSSW: CMSSW_11_1_X_2020-04-16-2300
SCRAM_ARCH: slc7_amd64_gcc820
You can see the results of the tests here:
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-9772ac/5744/summary.html

I found follow errors while testing this PR

**Failed tests:  Build  HeaderConsistency  ClangBuild**

* **Build**:

I found compilation error when building: 

<pre>>> Compiling  /data/cmsbld/jenkins/workspace/ib-run-pr-tests/CMSSW_11_1_X_2020-04-16-2300/src/DataFormats/Luminosity/src/LumiInfo.cc 
>> Compiling  /data/cmsbld/jenkins/workspace/ib-run-pr-tests/CMSSW_11_1_X_2020-04-16-2300/src/DataFormats/Luminosity/src/LumiDetails.cc 
>> Compiling  /data/cmsbld/jenkins/workspace/ib-run-pr-tests/CMSSW_11_1_X_2020-04-16-2300/src/DataFormats/Luminosity/src/LumiSummaryRunHeader.cc 
>> Building LCG reflex dict from header file src/DataFormats/Luminosity/src/classes.h
In file included from input_line_9:58:
/data/cmsbld/jenkins/workspace/ib-run-pr-tests/CMSSW_11_1_X_2020-04-16-2300/src/DataFormats/Luminosity/interface/PixelClusterCounts.h:37:20: error: no type named 'PixelClusterCountsInEvent' in namespace 'reco'
    void Add(reco::PixelClusterCountsInEvent pccInEvent) {
             ~~~~~~^
Error: rootcling: compilation failure (tmp/slc7_amd64_gcc820/src/DataFormats/Luminosity/src/DataFormatsLuminosity/a/DataFormatsLuminosity_xr2abd7024ae_dictUmbrella.h)
gmake: *** [tmp/slc7_amd64_gcc820/src/DataFormats/Luminosity/src/DataFormatsLuminosity/a/DataFormatsLuminosity_xr.cc] Error 1
>> Compiling LCG dictionary: tmp/slc7_amd64_gcc820/src/DataFormats/Luminosity/src/DataFormatsLuminosity/a/DataFormatsLuminosity_xr.cc
</pre>

* **Clang**:

I found compilation error while trying to compile with clang. Command used:
```
USER_CUDA_FLAGS='--expt-relaxed-constexpr' USER_CXXFLAGS='-Wno-register -fsyntax-only' scram build -k -j 32 COMPILER='llvm compile'

```


<pre>Entering library rule at src/Calibration/LumiAlCaRecoProducers/plugins
>> Compiling edm plugin /data/cmsbld/jenkins/workspace/ib-run-pr-tests/CMSSW_11_1_X_2020-04-16-2300/src/Calibration/LumiAlCaRecoProducers/plugins/AlcaPCCEventProducer.cc 
Entering library rule at src/Calibration/LumiAlCaRecoProducers/plugins
>> Compiling edm plugin /data/cmsbld/jenkins/workspace/ib-run-pr-tests/CMSSW_11_1_X_2020-04-16-2300/src/Calibration/LumiAlCaRecoProducers/plugins/RawPCCProducer.cc 
In file included from /data/cmsbld/jenkins/workspace/ib-run-pr-tests/CMSSW_11_1_X_2020-04-16-2300/src/Calibration/LumiAlCaRecoProducers/plugins/RawPCCProducer.cc:16:
/data/cmsbld/jenkins/workspace/ib-run-pr-tests/CMSSW_11_1_X_2020-04-16-2300/src/DataFormats/Luminosity/interface/PixelClusterCounts.h:37:20: error: no type named 'PixelClusterCountsInEvent' in namespace 'reco'; did you mean 'PixelClusterCounts'?
    void Add(reco::PixelClusterCountsInEvent pccInEvent) {
             ~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~
                   PixelClusterCounts
/data/cmsbld/jenkins/workspace/ib-run-pr-tests/CMSSW_11_1_X_2020-04-16-2300/src/DataFormats/Luminosity/interface/PixelClusterCounts.h:22:9: note: 'PixelClusterCounts' declared here
  class PixelClusterCounts {
</pre>

--- a/Calibration/LumiAlCaRecoProducers/plugins/AlcaPCCIntegrator.cc
+++ b/Calibration/LumiAlCaRecoProducers/plugins/AlcaPCCIntegrator.cc
@@ -10,8 +10,6 @@ ________________________________________________________________**/
 // C++ standard
 #include <string>
 // CMS
-#include "DataFormats/SiPixelCluster/interface/SiPixelCluster.h"
-#include "DataFormats/TrackerRecHit2D/interface/SiPixelRecHitCollection.h"
 #include "DataFormats/Luminosity/interface/PixelClusterCountsInEvent.h"
 #include "DataFormats/Luminosity/interface/PixelClusterCounts.h"
 

--- a/Calibration/LumiAlCaRecoProducers/plugins/AlcaPCCIntegrator.cc
+++ b/Calibration/LumiAlCaRecoProducers/plugins/AlcaPCCIntegrator.cc
@@ -86,7 +86,7 @@ void AlcaPCCIntegrator::produce(edm::Event& iEvent, const edm::EventSetup& iSetu
   iEvent.getByToken(pccToken_, pccHandle);
 
   const reco::PixelClusterCountsInEvent inputPcc = *(pccHandle.product());
-  thePCCob->Add(inputPcc);
+  thePCCob->add(inputPcc);
 }
 
 //--------------------------------------------------------------------------------------------------

--- a/DataFormats/Luminosity/interface/PixelClusterCounts.h
+++ b/DataFormats/Luminosity/interface/PixelClusterCounts.h
@@ -17,6 +17,7 @@
 #include "FWCore/MessageLogger/interface/MessageLogger.h"
 
 #include "DataFormats/Luminosity/interface/LumiConstants.h"
+#include "DataFormats/Luminosity/interface/PixelClusterCountsInEvent.h"
 
 namespace reco {
   class PixelClusterCounts {
@@ -34,17 +35,12 @@ namespace reco {
 
     void eventCounter(unsigned int bxID) { m_events.at(bxID - 1)++; }
 
-    void Add(reco::PixelClusterCountsInEvent pccInEvent) {
-      std::vector<int> countsInEvent = pccInEvent.counts();
-      std::vector<int> modIDInEvent = pccInEvent.modID();
+    void add(reco::PixelClusterCountsInEvent const& pccInEvent) {
+      std::vector<int> const& countsInEvent = pccInEvent.counts();
+      std::vector<int> const& modIDInEvent = pccInEvent.modID();
       int bxIDInEvent = pccInEvent.bxID();
       for (unsigned int i = 0; i < modIDInEvent.size(); i++) {
-        size_t modIndex = std::distance(m_ModID.begin(), std::find(m_ModID.begin(), m_ModID.end(), modIDInEvent.at(i)));
-        if (modIndex == m_ModID.size()) {
-          m_ModID.push_back(modIDInEvent.at(i));
-          m_counts.resize(m_counts.size() + LumiConstants::numBX, 0);
-        }
-        m_counts.at(LumiConstants::numBX * modIndex + bxIDInEvent - 1) += countsInEvent.at(i);
+        increment(modIDInEvent[i], bxIDInEvent, countsInEvent.at(i));
       }
     }
 

